### ISSUE TITLE
object-curly-spacing warns on trailing comments, fixer deletes

### ISSUE THREAD

**Issue by Slayer95** [eslint/eslint#11656]
**Tell us about your environment**

* **ESLint Version:** v5.15.1
* **Node Version:** v10.15.3
* **npm Version:** v6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	"root": true,
	"parserOptions": {
		"ecmaVersion": 9,
		"sourceType": "script",
		"ecmaFeatures": {
			"globalReturn": true
		}
	},
	"env": {
		"es6": true,
		"node": true
	},
	"rules": {
		"object-curly-spacing": ["error", "never"]
	}
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
"use strict";

function foo(data) {
	const {bar /* number */} = data;
}

```

```bash
eslint eslint-testcase.js
```

**What did you expect to happen?**
Lint passes.

**What actually happened? Please include the actual, raw output from ESLint.**
Rule ``object-curly-spacing`` detected a false positive.

C:\Users\W7\Desktop\osu-accuracy\eslint-testcase.js
  4:25  error  There should be no space before '}'  object-curly-spacing

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

Note that running with ``--fix`` deletes the comment ``/* number */``:

```diff
4c4
< 	const {bar /* number */} = data;
---
> 	const {bar} = data;
```

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by g-plane** [comment]
Hi @Slayer95 , thanks for this issue.

I can reproduce this bug.

**Comment by reintroducing** [comment]
I just upgraded to eslint 6.0.1 and this is still triggering a false positive for me:
`daytime_only: { // eslint-disable-line camelcase`

```
  ✘  https://google.com/#q=babel%2Fobject-curly-spacing  There should be no space after '{'
  ../../../../src/js/filters/filters-utils.js:42:19
```

**Comment by Slayer95** [comment]
@reintroducing , you are using ``babel/object-curly-spacing``.

Report at https://github.com/babel/eslint-plugin-babel

**Comment by reintroducing** [comment]
ugh, sorry, i'm getting lost in looking through issues in multiple repos.
